### PR TITLE
merge Busola's defaultConfig with ConfigMap

### LINKED ISSUE #1018: Merge configs properly on the backend
**Description**

In the backend, we either use the configuration from the `config.json` file or fully overwrite it by the configmap if it exists.  If we have a configmap, then we should merge the configs together, and still use the defaults for the values that have not been overwritten in the configmap.

**Expected result**

We should overwrite only those values that have been mentioned in the config map, not everything.

**Actual result**

We overwrite everything, we don't have any defaults if a configmap exists.

**Steps to reproduce**

**Troubleshooting**

### PR DESCRIPTION
**Pull request by wchrapka**
**Description**

Merge Busola's default configuration stored in a file with changes coming from Busola's ConfigMap

**Related issue(s)**

Fixes #1018

### CODE REVIEW

File: core/src/luigi-config/busola-cluster-params.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+        '/assets/defaultConfig.json' + cacheBuster,
+      );
+      const configMapResponse = await fetch(
+        '/assets/config/config.json' + cacheBuster,
+      );
+
+      const defultParams = await defaultConfigResponse.json();
+      const mapParams = await configMapResponse.json();
+
+      params = merge(defultParams, mapParams);
```
Comment (Wawrzyn321) [inline core/src/luigi-config/busola-cluster-params.js:19]: const defaultParams = await defaultConfigResponse.json();
      const mapParams = await configMapResponse.json();

      params = merge(defaultParams, mapParams);

**Review by Wawrzyn321 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
 
 Busola configuration is the product of gathering and merging the configurations from several individual sources. The following list presents the sources in the order of precedence:
 
+**Backend:**
+
+- Busola backend default cluster configuration, acquired from the file [defaultConfig.json](backend/settings/defaultConfig.json).
+- Busola cluster configuration, available on the Busola cluster in ConfigMap "busola/busola-config" under the key "config".
```
Comment (grego952) [inline README.md:60]:
```suggestion
- Busola cluster configuration, available on the Busola cluster in Config Map "busola/busola-config" under the key "config".
```

File: README.md [hunk 3]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
...
+  This data is mounted to the Busola `web` and `backend` pods, and during the local development,
```
Comment (grego952) [inline README.md:61]:
```suggestion
  This data is mounted to the Busola `web` and `backend` Pods, and during the local development,
```

File: README.md [hunk 4]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
...
+  the [defaultConfig.json](backend/settings/defaultConfig.json) file is used.
+
+**Frontend:**
+
 - Built-in, hardcoded defaults.
-- Busola cluster configuration, available on the Busola cluster in ConfigMap "busola/busola-config" under the key "config". This data is mounted to the Busola `web` pod, and during the local development, the [config.json](core/src/assets/config/config.json) file is used.
+- Busola frontend default cluster configuration, acquired from the file [defaultConfig.json](core/src/assets/defaultConfig.json).
+- Busola cluster configuration, available on the Busola cluster in ConfigMap "busola/busola-config" under the key "config".
```
Comment (grego952) [inline README.md:68]:
```suggestion
- Busola cluster configuration, available on the Busola cluster in Config Map "busola/busola-config" under the key "config".
```

File: README.md [hunk 5]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
...
+  This data is mounted to the Busola `web` and `backend` pods, and during the local development,
```
Comment (grego952) [inline README.md:69]:
```suggestion
  This data is mounted to the Busola `web` and `backend` Pods, and during the local development,
```

File: docs/features.md [hunk 6]
```diff
@@ -129,6 +129,8 @@ The document lists and describes all the feature flags that are available in Kym
 
 - **JWT_CHECK_CONFIG** – is used to configure data necessary for the backend authentication, such as an issuer and JWKS (JSON Web Key Set) address. When the feature is disabled no authentication occurs on backend side.
 
+  Backend feature. Cannot be modified at the cluster's ConfigMap level.
```
Comment (grego952) [inline docs/features.md:132]:
```suggestion
  Backend feature. Cannot be modified at the cluster's Config Map level.
```

File: docs/features.md [hunk 7]
```diff
@@ -143,6 +145,8 @@ The document lists and describes all the feature flags that are available in Kym
 
 - **GZIP** – is used to indicate whether the response from the backend server should be compressed or not.
 
+  Backend feature. Cannot be modified at the cluster's ConfigMap level.
```
Comment (grego952) [inline docs/features.md:148]:
```suggestion
  Backend feature. Cannot be modified at the cluster's Config Map level.
```

**Review by Wawrzyn321 — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
 
 Busola configuration is the product of gathering and merging the configurations from several individual sources. The following list presents the sources in the order of precedence:
 
+**Backend:**
+
+- Busola backend default cluster configuration, acquired from the [defaultConfig.json](backend/settings/defaultConfig.json) file.
+- Busola cluster configuration, available on the Busola cluster in Config Map "busola/busola-config" under the key "config".
```
Comment (grego952) [inline README.md:60]:
```suggestion
- Busola cluster configuration, available on the Busola cluster in the Config Map "busola/busola-config" under the key "config".
```

File: README.md [hunk 9]
```diff
@@ -54,8 +54,20 @@ Busola is delivered with the following default settings:
...
+  This data is mounted to the Busola `web` and `backend` Pods, and during the local development,
+  the [defaultConfig.json](backend/settings/defaultConfig.json) file is used.
+
+**Frontend:**
+
 - Built-in, hardcoded defaults.
-- Busola cluster configuration, available on the Busola cluster in ConfigMap "busola/busola-config" under the key "config". This data is mounted to the Busola `web` pod, and during the local development, the [config.json](core/src/assets/config/config.json) file is used.
+- Busola frontend default cluster configuration, acquired from the [defaultConfig.json](core/src/assets/defaultConfig.json) file.
+- Busola cluster configuration, available on the Busola cluster in Config Map "busola/busola-config" under the key "config".
```
Comment (grego952) [inline README.md:68]:
```suggestion
- Busola cluster configuration, available on the Busola cluster in the Config Map "busola/busola-config" under the key "config".
```

**Review by grego952 — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**